pkg/config/generate_test.go

```diff
@@ -97,3 +221,60 @@ func TestSelectDNSNames(t *testing.T) {
 
 	}
 }
+
+func testOpenShiftCluster() *api.OpenShiftManagedCluster {
```

Comment:
testOpenShiftClusterYAML above took the tack of a small yaml representation and unmarshalling it.  This PR is adding two further testOpenShiftCluster() functions.  Can this be simplified?

Reply:
the reason is a lack of deepcopy implementation, will note to fix in #192
nonetheless can the two testOpenShiftCluster() functions be condensed?

Reply:
sure, it needs to be a third package though so we don't make weird dependencies.  I'll throw it as a fixture.  Having it not be in a test package is also a smell to me though, I'd rather accept duplication in the unit test and allow the creation be specific to the package itself since, strictly speaking, testing different plugin impl methods represent different stages of the config's lifecycle.

Done.